src/problem.jl

```diff
@@ -80,14 +80,14 @@ Struct representing a closed-loop neural-network controlled system.
 """
 struct ControlledPlant{ST, XT, DT, PT, CNT, CPT} <: AbstractNeuralNetworkControlProblem
     ivp::InitialValueProblem{ST, XT}
-    controller::Network
+    controller
     vars::Dict{Symbol, DT}
     period::PT
     normalization::CNT
     preprocessing::CPT
 
     function ControlledPlant(ivp::InitialValueProblem{ST, XT},
-                             controller::Network,
+                             controller,
```

Comment:
This is the only "downgrade" in this PR. I had to find a way to define a controller that is not a `Network`. So I would just make this another type parameter.